extensions/amp-mega-menu/validator-amp-mega-menu.protoascii

```diff
@@ -0,0 +1,188 @@
+#
+# Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+#
+tags: {  # amp-mega-menu
+  html_format: AMP
+  tag_name: "SCRIPT"
+  extension_spec: {
+    name: "amp-mega-menu"
+    version: "0.1"
+    version: "latest"
+  }
+  attr_lists: "common-extension-attrs"
+}
+tags: {  # <amp-mega-menu>
+  html_format: AMP
+  tag_name: "AMP-MEGA-MENU"
+  requires_extension: "amp-mega-menu"
+  requires: "amp-mega-menu nav"
+  attr_lists: "extended-amp-global"
+  spec_url: "https://amp.dev/documentation/components/amp-mega-menu"
+  descendant_tag_list: "amp-mega-menu-allowed-descendants"
+  child_tags: {
+    mandatory_num_child_tags: 1
+    child_tag_name_oneof: "NAV"
+    child_tag_name_oneof: "AMP-LIST"
```

Comment:
Can you add a test for this? amp-list does not satisfy "amp-mega-menu" so given the `mandatory_num_child_tags: 1`, I don't see how this can be used (unless you have two mega-menus as I mention above).

Reply:
So my intention is that mega menu can follow one of two patterns:
`<amp-mega-menu>` > `<nav>` > ... or
`<amp-mega-menu>` > `<amp-list>` > `<template>` > `<nav>` > ...
Where the `nav` underneath `<template>` would satisfy the requirement. Examples of both cases are included in `validator-amp-mega-menu.html`.